test/task.js

```diff
@@ -64,14 +71,18 @@ describe('task', function() {
   });
 
   it('should get a task that was registered', function(done) {
-    taker.task('test1', noop);
+    expect(taker.task('test1', noop)).to.be.undefined();
+    expect(taker.task('test1')).to.equal(noop);
+    expect(taker.task('test1')).to.equal(noop);
     expect(taker.task('test1')).to.equal(noop);
     done();
   });
 
   it('should return a function that was registered in some other way', function(done) {
-    taker.registry()._tasks.test1 = noop;
+    taker.task('test1', noop);
```

Comment:
This is mixed the change for the previous modifications.
I'll modify back to original.